share/wake/lib/system/remote_cache_runner.wake

```diff
@@ -0,0 +1,210 @@
+            # --- Search the cache for the job  ---
+            # -------------------------------------
+
+            # Search the cache for a match
+            require Pass response =
+                rscApi
+                | rscApiFindMatchingJob (mkSearchRequest input hashKey)
+            else
+                # Leave a breadcrumb and fallback to the base runner if the cache fails to respond
+                def _ = breadcrumb "Failed to search for job in the cache"
+
+                # This job isn't getting cached. That's probably preferable since the server
+                # request failed but good to keep in mind.
+                baseDoIt job (Pass input)
+
+            # If a match was found use it
+            require NoMatch = response
+            else
+                require (Match details) = response
+                else unreachable "two-constructor tuple must have one value"
+
+                def _ = breadcrumb "Found a match in the cache"
+
+                def (
+                    CacheSearchResponseMatch
+                    outputSymlinks
+                    outputDirs
+                    outputFiles
+                    stdoutBlob
+                    stderrBlob
+                    status
+                    runtime
+                    cputime
+                    mem
+                    ibytes
+                    obytes
+                ) = details
+
+                def doDownload ((CacheSearchOutputFile path mode blob): CacheSearchOutputFile) =
+                    rscApiGetFileBlob blob "{input.getRunnerInputDirectory}/{path}" mode
+
+                def blobDownloads =
+                    outputFiles
+                    | map doDownload
+
+                def stdoutDownload = rscApiGetStringBlob stdoutBlob
+                def stderrDownload = rscApiGetStringBlob stderrBlob
+
+                require Pass stdout = stdoutDownload
+                else failWithError "rsc: Failed to download stdout"
+
+                require Pass stderr = stderrDownload
+                else failWithError "rsc: Failed to download stderr"
+
+                # We don't actually care about the result here but we do need to ensure that all
+                # downloads have completed and succeeded before we continue on.
+                require Pass _ =
+                    blobDownloads
+                    | findFail
+                else failWithError "rsc: Failed to download a blob"
+
+                def resolvedOutputs =
+                    outputFiles
+                    | map (_.getCacheSearchOutputFilePath)
+
+                def outputs = resolvedOutputs ++ outputDirs ++ outputSymlinks
+                def predict = Usage status runtime cputime mem ibytes obytes
+                def _ = virtual job stdout stderr status runtime cputime mem ibytes obytes
+                def inputs = map getPathName (input.getRunnerInputVisible)
+
+                Pass (RunnerOutput inputs outputs predict)
```

Comment:
Some infra outside of this is turning these string paths into `Path`s?

Reply:
Yeah, so the runner has to work entirely in `String`s because the `Job` is responsible for listing which `String`s should be elavated to `Path`s via `FnOutputs`

Since the rscRunner is just putting the stuff in the right place on disk its still up to the enclosing job to list which files should become `Path`s